### DISCUSSION TITLE
Call for maintainers

### DISCUSSION THREAD

**Discussion by martin-schulze-vireso** [bats-core/bats-core#1023] [Announcements]
You probably noticed the slowdown of Bats development in the last year.

My available free time has reduced and the chores have worn my motivation down.
This has considerably reduced my capacity to work on Bats and has lead to missed milestone deadlines.
I am not (yet?) stepping down as a maintainer but I want to inform you that this situation is not likely to improve soon.

As I don't want to see this project's development freeze (again), I am calling for (new) maintainers to step up and help bring new momentum to the development of Bats.

If you are interested in becoming a maintainer, post your "application" below, containing:

- what motivates you to maintain Bats
- where you think you can have the biggest impact for this project
- links to some of your bash projects or PRs to Bats
- anything else you want us to know

**Comment by brokenpip3** [comment]
I’m sad  to hear that you’re struggling, but i completely understand the challenges of limited time and motivation. I’ve been grappling with similar feelings while working another oss project, it’s difficult to consistently maintain high standards and find the energy to sustain both a project and its community.
That said, I want **to take this opportunity to acknowledge the incredible work you’ve done over the years** to revive, maintain, and introduce powerful, shiny new features to bats and its ecosystem. everyone in this community owes you a huge debt of gratitude for the energy and dedication you’ve poured into this project, and I'’m confident many others share the same.

From my side, I’ll continue to maintain the github action, homebrew packages, and docker image. However, lacks the deep knowledge of bats’ internals and, more importantly, due to very limited free time I'm unable to step into the role of an all-around bats contributor. 
I hope more people will step up in response to this call so we can collectively keep the bats organization vibrant.
 
At the same time, yet again thanks to martin's work, bats is not in bad shape, and the maintenance burden doesn’t seem overwhelming at the moment, so don't be scared, we can work together :)
Luigi

**Comment by jeff-cook** [comment]
I love bats and find it to be a very useful tool. It's been a few years/jobs since it has been the right tool, but I'm back to using it for a lot of my testing needs. 

I'm a systems/devops engineer. So a lot of my tools are bash first. 

I also use bats a wrapper around other tools to be able to add insights and reporting into their out, typically in a ci pipeline. 

I don't have much public code I can point you at, but I'm interested in helping you maintain this project.

**Reply by martin-schulze-vireso** [reply]
Thanks for your post Jeff. Maybe we can compensate for the lack of public portfolio by creating it just in time? Unfortunately, there are no coding related issues marked as `good first issue` right now, but maybe you can find [small](https://github.com/bats-core/bats-core/issues?q=is%3Aopen+is%3Aissue+label%3A%22Size%3A+Small%22) or [medium sized](https://github.com/bats-core/bats-core/issues?q=is%3Aopen+is%3Aissue+label%3A%22Size%3A+Medium%22) issues to create a PR for?

You could also look through the recent existing PRs and leave a review there?

**Comment by Wuodan** [comment]
Hi,

I'm interested in becoming a maintainer of Bats, and I've already been thinking about how to contact you for this.

Bats has been in several of my projects since early 2026.

I already created an org [bats-hardened](https://github.com/bats-hardened) to hold my forks of bats repos together and am opening PRs.

## What motivates you to maintain Bats?

- I like the project.
  - Has the features it needs. Best out there for the task!
  - It's mature, many people put effort into it over ~15 years.
  - It supports installation across a range of environments and tests those paths well.
  - It works well in my own projects.
- The project could use some polishing:
  - bats-action has _issues_. (Cache broken, feature creep, sudo-install, not guaranteed reproducible in toolchains because it references movable git-tags, ...)
  - The maintenance workload appears manageable as the project is already mature.
- I have the time and skills.


## Where do you think you can have the biggest impact for this project?

These are suggestions. I expect this to be coordinated with other maintainers, unless they want/have to step back due to lack of time.

1. Initial polish
   - refactor bats-action: Mostly done in my fork at [bats-hardened/bats-action](https://github.com/bats-hardened/bats-action)
   - Fix tests and hiccups in other Bats repositories; this work is ongoing in my forks.
   - Add linters to test workflows in each repo. I have two setups from my own projects
     - small-fast (super-linter): fast enough to be included in automated test workflows
     - big-slow (megalinter): drop-in workflow for manual runs (and optional self-hosted). Very picky. Initial effort until repo is linted, then run periodically.
   - Review open issues and PRs for overlooked improvements and approachable contributions.
2. Ongoing maintenance
   - Dependency maintenance: Test setup is already good, after polishing I expect the needed effort for dependency maintenance to be low.
     1. Weekly [look at all open PRs by dependabot](https://github.com/pulls/search?q=is%3Apr+is%3Aopen+org%3Abats-core+sort%3Aupdated-desc+author%3Adependabot%5Bbot%5D)
     2. Merge when tests pass – trust the strong test foundation.
   - Respond to user PRs and issues. Pick up and test/tweak new ideas by users.
   - Create a release when:
     - other maintainers agree ;-)
     - dependency updates demand it (security critical updates or many small updates since last release - tbd)

## Links to some of your bash projects or PRs to Bats

- [pentoo/pentoo-installer](https://github.com/pentoo/pentoo-installer)
  - Installer of the [Pentoo](https://en.wikipedia.org/wiki/Pentoo) Linux distro written in bash.
  - Written by me in 2014 and still in use.
- My PRs to Bats can be seen [here](https://github.com/pulls/search?q=is%3Apr+author%3AWuodan+sort%3Aupdated-desc+org%3Abats-core).
  The list is growing; at the moment, it contains seven open PRs.
- My forks of Bats repositories are collected in [bats-hardened](https://github.com/bats-hardened).
- While few of my projects are written exclusively in `bash`, nearly all of them use it for workflows, container builds, and related automation. Recent examples include:
  - [aicage](https://github.com/aicage/aicage): the `aicage` organization contains five repositories that build and maintain a chain of container images, with extensive tests and CI.
  - [ghcr-cleanup-manager](https://github.com/ghcr-manager/ghcr-cleanup-manager): a tool and GitHub Action for cleaning up old images and visualizing image-registry relationships.

## Anything else you want us to know?

I am Swiss, live near Zurich, started programming in 1985/86, and have worked as a professional software developer since 2000.

If you would like to contact me directly, I can send my phone number by email. I speak German and English, as well as some French.

**Reply by martin-schulze-vireso** [reply]
Hello @Wuodan, thanks for your post and PRs. I will have a look at them as soon as I find some time to. I would also like to encourage you to review other PRs and give suggestions there, so we can see what your work on the "other side" looks like.

**Reply by brokenpip3** [reply]
ehi @Wuodan thanks for your time and enthusiasm :)

>refactor bats-action

If you have any suggestion there we are super happy to accept PRs or work in pair.
Especially if you have any quick wins that we can apply to the current code structure :)

For a complete refactor let's open an issue and discuss together how to tackle it

**Reply by Wuodan** [reply]
@brokenpip3 By coincidence I just created the PR for bats-action that I prepared. It's a complete refactor as doing things in steps would add a lot of overhead on me and reviewers. If you see a sane way to split this please let me know.

**Reply by brokenpip3** [reply]
> @brokenpip3 By coincidence I just created the PR for bats-action that I prepared. It's a complete refactor as doing things in steps would add a lot of overhead on me and reviewers. If you see a sane way to split this please let me know.

yup just saw it, let's discuss there :)

edit: will take a complete look asap
